circle.yml

```diff
@@ -7,445 +6,305 @@ orbs:
-                      docker build -t cypress/included:<< parameters.dockerTag >> --platform << parameters.platformArg >> .
-                  working_directory: included/<< parameters.dockerTag >>
-
-            - test-included-image-versions:
-                  cypressVersion: << parameters.dockerTag >>
-                  imageName: cypress/included:<< parameters.dockerTag >>
-
-            - test-included-image:
-                  cypressVersion: << parameters.dockerTag >>
-                  imageName: cypress/included:<< parameters.dockerTag >>
+                      if [ << parameters.target >> != factory ]; then docker-compose build --progress plain factory; fi
+                      docker-compose build --progress plain << parameters.target >>
+                  working_directory: factory
+            - run:
+                  name: test
+                  command: |
+                    docker-compose build --progress plain << parameters.test-target >>
+                    docker-compose run << parameters.test-target >>
+                  working_directory: factory/test-project
 
-    push-images:
-        ## the multi-arch images have been built and tested by previous jobs, so we're safe to blindly build+push
+    push:
         machine:
-            image: ubuntu-2004:202111-02
+            image: ubuntu-2204:2022.10.2
         parameters:
-            dockerName:
-                type: string
-                description: Image name to build
-            dockerTag:
+            target:
                 type: string
-                description: Image tag to build, should match Cypress version, like "3.8.1"
-            workingDirectory:
-                type: string
-                description: Directory with the Dockerfile
+                description: Name of the docker-compose target to build and push.
         steps:
             - checkout
+            - expand-env-file
             - halt-if-docker-image-exists:
-                imageName: << parameters.dockerName >>:<< parameters.dockerTag >>
-            - halt-on-branch
+                 target: << parameters.target >>
             - run:
-                name: build + push multi-arch image
-                working_directory: << parameters.workingDirectory >>
+                name: building Docker image for target << parameters.target >>
                 command: |
-                    ## install QEMU, pulled from https://github.com/docker/setup-qemu-action/blob/4620c11b701e878e3f2b7aa142927d0646857fd9/src/main.ts
-                    ## avoids https://github.com/docker/buildx/issues/499
-                    ## this allows us to build for arm64 from x86_64 in CI, by installing QEMU utils and binfmt data on our Docker host
-                    ## if this suddenly breaks, try a newer digest version or the docker/binfmt fork
-                    BINFMT_IMAGE=tonistiigi/binfmt:latest@sha256:85683def11494bc0055d98dd1081e73cb5e70bbc1ae532be3d12cf054e2b11f1
```

Comment:
why did we change this from a specific image to be open as `tonistiigi/binfmt`?

Reply:
Since I updated the machine version we no longer have to work around the specified issue.